src/Analyzers/CSharp/Analyzers/UseCollectionExpression/UseCollectionExpressionHelpers.cs

```diff
@@ -265,14 +272,22 @@ bool IsSafeConversionWhenTypesDoNotMatch(out bool changesSemantics)
         }
 
         bool IsWellKnownInterface(ITypeSymbol type)
+            => IsWellKnownReadOnlyInterface(type) || IsWellKnownReadWriteInterface(type);
+
+        bool IsWellKnownReadOnlyInterface(ITypeSymbol type)
         {
             return type.OriginalDefinition.SpecialType
                 is SpecialType.System_Collections_Generic_IEnumerable_T
-                or SpecialType.System_Collections_Generic_ICollection_T
-                or SpecialType.System_Collections_Generic_IList_T
                 or SpecialType.System_Collections_Generic_IReadOnlyCollection_T
                 or SpecialType.System_Collections_Generic_IReadOnlyList_T;
         }
+
+        bool IsWellKnownReadWriteInterface(ITypeSymbol type)
```

Comment:
Maybe mutable or writable?

```suggestion
        bool IsWellKnownMutableInterface(ITypeSymbol type)
```

Reply:
i'm ok with this name :)